packages/block-editor/src/components/block-list/index.js

```diff
@@ -140,9 +144,13 @@ function StopEditingAsBlocksOnOutsideSelect( { clientId } ) {
 	);
 	useEffect( () => {
 		if ( ! isBlockOrDescendantSelected ) {
-			stopEditingAsBlocks( clientId );
```

Comment:
The clientId was incorrectly passed, the selector takes no params.

Reply:
`stopEditingAsBlocks` did take a clientId though: https://github.com/WordPress/gutenberg/pull/72959/files#diff-4d145a0c43689f5a56c00a8e2919550768ebd233e91a6c6d04fee71a44bae9f2L332

`stopEditingContentOnlySection` is nicer in its simplicity.